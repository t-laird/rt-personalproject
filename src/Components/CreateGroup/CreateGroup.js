import React, { Component } from 'react';
import { connect } from 'react-redux';
import generator from 'generate-password';
import * as actions from '../../Actions';

class CreateGroup extends Component {
	constructor() {
		super();

		this.state = {
			groupName: '',
			weeklyPoints: '',
			message: 'Create a new group'
		}
	}

	createGroup = async (event, groupName, weeklyPoints) => {
		event.preventDefault();
		const password = generator.generate({
			length: 6,
			number: true
		})

		const response = await fetch('http://localhost:3000/api/v1/group/new', {
			method: 'POST',
			body: JSON.stringify({
				group_name: groupName,
				group_passphrase: password,
				weekly_points: weeklyPoints,
				administrator_id: this.props.user.user_id
			}),
			headers: {
				'Content-Type': 'application/json'
			}
		})

		const cleanResponse = await response.json();
		console.log('1234', response, cleanResponse)
		this.props.updateGroup(cleanResponse[0])
	}

	handleChange = (event) => {
		const { target } = event;
		const { name, value } = target;

		this.setState({[name]: value})
	}

	render() {
		return (
			<div>
				<h4>{this.state.message}</h4>
				<form>
					<input 
						onChange={this.handleChange}
						placeholder='choose group name'
						value={this.state.groupName}
						name='groupName'
					/>
					<input 
						onChange={this.handleChange}
						placeholder='choose number of weekly points members receive'
						value={this.state.weeklyPoints}
						name='weeklyPoints'
					/>
					<button
						onClick={(event) => this.createGroup(event, this.state.groupName, this.state.weeklyPoints)}
					>submit
					</button>
				</form>
			</div>
		)
	}
}

const mapStateToProps = ( store ) => ({
	user: store.User
});

const mapDispatchToProps = dispatch => ({
	updateGroup: group => {
		dispatch(actions.updateGroup(group))
	}
});



export default connect(mapStateToProps, mapDispatchToProps)(CreateGroup);