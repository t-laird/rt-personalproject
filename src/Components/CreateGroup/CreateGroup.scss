@import '../../Variables.scss';

.create-group-component {
	@include center-box;
	background-color: $c6;

	.create-group-heading {
		@include box-header;
		background-color: $c1;

		h4 {
			color: $c7;
			font-weight: 900;
			text-align: center;
		}
	}

	h5 {
		font-weight: 900;
		color: $c3;
		line-height: 1.3;
		margin-bottom: 20px;
	}

	.success-message {
		margin-top: -19px;
		margin-bottom: 26px;
		
		h5 {
			color: $c1;
		}

		.success-span {
			color: $c3;
			font-size: 28px;
			margin-left: 5px;
		}
	}

	form {
		padding: 24px 40px;
	}

	span {
		font-weight: 900;
		font-family: 'semplicitapro', sans-serif;
		font-size: 20px;
		color: $c4;
	}

	input {
		background-color: $c7;
		color: $c3;
		border: none;
		border-bottom: 8px solid $c5;

		&::placeholder {
			font-weight: 300;
			font-size: 24px;
			bottom: 2px;
		}
	}

	input, button {
		height: 60px;
		font-size: 24px;
		text-align: center;
		margin-top: 6px;
	}

	.group-name-input {
		margin-bottom: 15px;
		margin-left: 10px;
	}

	.points-input {
		width: 106px;
		font-size: 34px;
		margin-left: 30px;

		&::placeholder {
			position: relative;
			bottom: 3px;
		};
	}

	.create-top, .create-bottom {
		display: flex;
		justify-content: space-between;
		margin-bottom: 20px;

		input {
			outline: none;

			&::placeholder {
				color: $c4;
				font-weight: 300;
			};
		}
	}

	button {
		height: 52px;
		width: 149px;
		color: #fff;
		background-color: $c1;
		cursor: pointer;
		transition: background-color .15s ease-in-out;
		padding-top: 2px;
		margin: 41px auto 0;
		border: none;
		display: block;
		font-size: 20px;

		&:hover {
      background-color: darken($c1, 12%);
    }
	}
}

.link-holder .create-group-link {
	margin-bottom: 60px;
}

@media screen and (max-width: 768px) {
	.create-group-component {
		@include media-width;

		.create-group-heading {
			@include media-width;
		}
	}
}

@media screen and (max-width: 555px) {
	.create-group-component {
		.create-top {
			flex-direction: column;
			width: 100%;

			input {
				margin-left: 0;	
			}
		}
	}
}

@media screen and (max-width: 555px) {
	.create-group-component {
		.create-bottom {
			flex-direction: column;
			width: 100%;

			input {
				width: 100%;
				margin-left: 0;
			}
		}
	}
}