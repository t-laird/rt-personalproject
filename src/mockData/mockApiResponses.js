export const getGroupSettingsApiResponse = {
  status: 'success',
};

export const getGroupTransactionDataApiResponse = {
	status: 'success'
};