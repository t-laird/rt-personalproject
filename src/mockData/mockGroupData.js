export const mockGroupData = {
	administrator_id: 95,
	created_date: "2017-10-18T18:34:30.017Z",
	group_id: 1,
	group_name: 'Turing',
	group_passphrase: "9139z",
	weekly_points: 100
}