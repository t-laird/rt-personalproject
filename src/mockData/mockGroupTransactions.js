const mockGroupTransactions = [
  {transactions: [
    {
      created_time: 1514939019586,
      event_id: 9,
      event_time: "2018-01-03T00:23:39.627Z",
      group_id: 1,
      note: "You totally killed it on your lightning talk this morning!  I loved it.",
      point_value: 15,
      receive_id: 22,
      received_name: "Scott Adams",
      send_id: 24,
      send_name: "Jim Davis"
    },
    {
      created_time: 1514939019586,
      event_id: 2,
      event_time: "2018-01-03T00:23:39.612Z",
      group_id: 1,
      note: "Thanks!",
      point_value: 10,
      receive_id: 59,
      received_name: "Bo Jackson",
      send_id: 43,
      send_name: "Deion Sanders"
    }
  ]}
]

export default mockGroupTransactions;