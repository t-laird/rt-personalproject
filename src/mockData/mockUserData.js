export const mockUserData = {
  authrocket_id: "usr_0vXWX9jSRkbUjLGsOb9S8r",
  created_date: "2017-11-18T18:34:30.017Z",
  email: "cpavin@usopen.com",
  group_id: 1,
  name: "Corey Pavin",
  user_id: 95
};

