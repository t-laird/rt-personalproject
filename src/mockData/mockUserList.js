export const mockUserList = [
  {
    authrocket_id: "jfnxcio42393",
    created_date: "2018-01-03T01:39:33.544Z",
    email: "abcf@gmail.com",
    group_id: 1,
    name: "Mike Jenkins",
    user_id: 2
  },
  {
    authrocket_id: "lksdfjn23413",
    created_date: "2018-01-03T01:39:33.544Z",
    email: "qwi123@gmail.com",
    group_id: 1,
    name: "Josh Brown",
    user_id: 3
  },
  {
    authrocket_id: "xmcv3941nv",
    created_date: "2018-01-03T01:39:33.545Z",
    email: "293xcnv@gmail.com",
    group_id: 1,
    name: "Alan Michaelson",
    user_id: 4
  },
  {
    authrocket_id: "3490xxa1j",
    created_date: "2018-01-03T01:39:33.545Z",
    email: "nc-1n23a@gmail.com",
    group_id: 1,
    name: "Sarah Hilman",
    user_id: 5    
  } 
];