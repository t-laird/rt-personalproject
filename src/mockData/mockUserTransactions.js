const mockUserTransactions = [
  {
    received: [
      {
        'send_id': 5,
        'receive_id': 8,
        'group_id': 1,
        'created_time': Date.now(),    
        'point_value': 10,
        'send_name': 'Ronald Marshall',
        'note': 'Tacos make me happy.'
      },
      {
        'send_id': 1,
        'receive_id': 8,
        'group_id': 1,
        'created_time': 1504239487190,
        'point_value': 10,
        'send_name': 'Paavo Marsicek',
        'note': 'I did not think we would ever get finished with testing.'  
      }
    ],
    sent: [
      {
        'send_id': 8,
        'receive_id': 4,
        'group_id': 1,
        'created_time': Date.now(),    
        'point_value': 10,
        'send_name': 'Ronald Marshall',
        'note': 'Tacos make me happy.'
      },
      {
        'send_id': 8,
        'receive_id': 3,
        'group_id': 1,
        'created_time': 1504239487190,
        'point_value': 10,
        'send_name': 'Paavo Marsicek',
        'note': 'I did not think we would ever get finished with testing.'  
      }
    ]
  }
];

export default mockUserTransactions;

