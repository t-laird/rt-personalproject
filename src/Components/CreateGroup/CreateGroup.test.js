import { CreateGroup, mapStateToProps, mapDispatchToProps } from './CreateGroup';
import React from 'react';
import { shallow } from 'enzyme';
import { mockUserData } from '../../mockData/mockUserData';
import { mockGroupData } from '../../mockData/mockGroupData';

global.localStorage = {
  getItem: () => null
};

describe('Create group tests', () => {
	it('should match the snapshot', () => {
		const renderedApp = shallow(<CreateGroup user={{}} />)

		expect(renderedApp).toMatchSnapshot();
	})

	it('should set state on input', () => {
		const renderedApp = shallow(<CreateGroup user={{}} />)
		const mockEvent = { target: { name: 'groupName', value: 'def' } }

		renderedApp.instance().handleChange(mockEvent)
		expect(renderedApp.state('groupName')).toEqual('def');
	})

	it('should call updateGroup action when createGroup is invoked', async () => {
		window.fetch = jest.fn().mockImplementation(() =>
	    Promise.resolve({
	      json: () => Promise.resolve([mockGroupData])
	    })
	  );

		const mockUpdateGroup = jest.fn();
		const mockUpdateUser = jest.fn();
		const renderedApp = shallow(<CreateGroup user={{}} updateUser={mockUpdateUser} updateGroup={mockUpdateGroup} />)
		const mockEvent = { preventDefault: jest.fn() }
		const mockGroupName = 'Turing';
		const mockWeeklyPoints = 50;

		await renderedApp.instance().createGroup(mockEvent, mockGroupName, mockWeeklyPoints);
		renderedApp.update();

		expect(mockUpdateGroup).toHaveBeenCalledWith(mockGroupData);
	})

	it('should call updateUser action when createGroup is invoked', async () => {
		window.fetch = jest.fn().mockImplementation(() =>
	    Promise.resolve({
	      json: () => Promise.resolve([mockUserData])
	    })
	  );

		const mockUpdateGroup = jest.fn();
		const mockUpdateUser = jest.fn();
		const renderedApp = shallow(<CreateGroup user={{}} updateUser={mockUpdateUser} updateGroup={mockUpdateGroup} />)
		const mockEvent = { preventDefault: jest.fn() }
		const mockGroupName = 'Turing';
		const mockWeeklyPoints = 50;

		await renderedApp.instance().createGroup(mockEvent, mockGroupName, mockWeeklyPoints);
		renderedApp.update();

		expect(mockUpdateUser).toHaveBeenCalledWith(mockUserData);
	})
})

describe('mapStateToProps tests', () => {

})

describe('mapDispatchToProps tests', () => {
	
})

