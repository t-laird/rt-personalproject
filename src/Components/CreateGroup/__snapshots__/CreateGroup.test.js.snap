// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create group tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateGroup
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="create-group-heading"
>
          <h4>
                    CREATE A GROUP
          </h4>
</div>,
        <form>
          <div
                    className="create-top"
          >
                    <h5>
                              Let's give your group a name!
                    </h5>
                    <input
                              className="group-name-input"
                              name="groupName"
                              onChange={[Function]}
                              placeholder="group name"
                              value=""
                    />
          </div>
          <h5>
                    With SNAP NINJA, your group members get a bucket of snaps each week to give away to deserving teammates.
          </h5>
          <div
                    className="create-bottom"
          >
                    <h5>
                              How many snaps will your group members be able to give away?
                              <span
                                        className="join-span"
                              >
                                        (Between 1 - 500)
                              </span>
                    </h5>
                    <input
                              className="points-input"
                              name="weeklyPoints"
                              onChange={[Function]}
                              placeholder="QTY"
                              value=""
                    />
          </div>
          <div
                    className="success-message"
          />
          <button
                    onClick={[Function]}
          >
                    CREATE
          </button>
</form>,
      ],
      "className": "create-group-component",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h4>
            CREATE A GROUP
</h4>,
          "className": "create-group-heading",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "CREATE A GROUP",
          },
          "ref": null,
          "rendered": "CREATE A GROUP",
          "type": "h4",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="create-top"
>
              <h5>
                            Let's give your group a name!
              </h5>
              <input
                            className="group-name-input"
                            name="groupName"
                            onChange={[Function]}
                            placeholder="group name"
                            value=""
              />
</div>,
            <h5>
              With SNAP NINJA, your group members get a bucket of snaps each week to give away to deserving teammates.
</h5>,
            <div
              className="create-bottom"
>
              <h5>
                            How many snaps will your group members be able to give away?
                            <span
                                          className="join-span"
                            >
                                          (Between 1 - 500)
                            </span>
              </h5>
              <input
                            className="points-input"
                            name="weeklyPoints"
                            onChange={[Function]}
                            placeholder="QTY"
                            value=""
              />
</div>,
            <div
              className="success-message"
/>,
            <button
              onClick={[Function]}
>
              CREATE
</button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h5>
                  Let's give your group a name!
</h5>,
                <input
                  className="group-name-input"
                  name="groupName"
                  onChange={[Function]}
                  placeholder="group name"
                  value=""
/>,
              ],
              "className": "create-top",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Let's give your group a name!",
                },
                "ref": null,
                "rendered": "Let's give your group a name!",
                "type": "h5",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "group-name-input",
                  "name": "groupName",
                  "onChange": [Function],
                  "placeholder": "group name",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "With SNAP NINJA, your group members get a bucket of snaps each week to give away to deserving teammates.",
            },
            "ref": null,
            "rendered": "With SNAP NINJA, your group members get a bucket of snaps each week to give away to deserving teammates.",
            "type": "h5",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h5>
                  How many snaps will your group members be able to give away?
                  <span
                                    className="join-span"
                  >
                                    (Between 1 - 500)
                  </span>
</h5>,
                <input
                  className="points-input"
                  name="weeklyPoints"
                  onChange={[Function]}
                  placeholder="QTY"
                  value=""
/>,
              ],
              "className": "create-bottom",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "How many snaps will your group members be able to give away?",
                    <span
                      className="join-span"
>
                      (Between 1 - 500)
</span>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "How many snaps will your group members be able to give away?",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "(Between 1 - 500)",
                      "className": "join-span",
                    },
                    "ref": null,
                    "rendered": "(Between 1 - 500)",
                    "type": "span",
                  },
                ],
                "type": "h5",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "points-input",
                  "name": "weeklyPoints",
                  "onChange": [Function],
                  "placeholder": "QTY",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "success-message",
            },
            "ref": null,
            "rendered": "",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "CREATE",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "CREATE",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="create-group-heading"
>
            <h4>
                        CREATE A GROUP
            </h4>
</div>,
          <form>
            <div
                        className="create-top"
            >
                        <h5>
                                    Let's give your group a name!
                        </h5>
                        <input
                                    className="group-name-input"
                                    name="groupName"
                                    onChange={[Function]}
                                    placeholder="group name"
                                    value=""
                        />
            </div>
            <h5>
                        With SNAP NINJA, your group members get a bucket of snaps each week to give away to deserving teammates.
            </h5>
            <div
                        className="create-bottom"
            >
                        <h5>
                                    How many snaps will your group members be able to give away?
                                    <span
                                                className="join-span"
                                    >
                                                (Between 1 - 500)
                                    </span>
                        </h5>
                        <input
                                    className="points-input"
                                    name="weeklyPoints"
                                    onChange={[Function]}
                                    placeholder="QTY"
                                    value=""
                        />
            </div>
            <div
                        className="success-message"
            />
            <button
                        onClick={[Function]}
            >
                        CREATE
            </button>
</form>,
        ],
        "className": "create-group-component",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h4>
              CREATE A GROUP
</h4>,
            "className": "create-group-heading",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "CREATE A GROUP",
            },
            "ref": null,
            "rendered": "CREATE A GROUP",
            "type": "h4",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="create-top"
>
                <h5>
                                Let's give your group a name!
                </h5>
                <input
                                className="group-name-input"
                                name="groupName"
                                onChange={[Function]}
                                placeholder="group name"
                                value=""
                />
</div>,
              <h5>
                With SNAP NINJA, your group members get a bucket of snaps each week to give away to deserving teammates.
</h5>,
              <div
                className="create-bottom"
>
                <h5>
                                How many snaps will your group members be able to give away?
                                <span
                                                className="join-span"
                                >
                                                (Between 1 - 500)
                                </span>
                </h5>
                <input
                                className="points-input"
                                name="weeklyPoints"
                                onChange={[Function]}
                                placeholder="QTY"
                                value=""
                />
</div>,
              <div
                className="success-message"
/>,
              <button
                onClick={[Function]}
>
                CREATE
</button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h5>
                    Let's give your group a name!
</h5>,
                  <input
                    className="group-name-input"
                    name="groupName"
                    onChange={[Function]}
                    placeholder="group name"
                    value=""
/>,
                ],
                "className": "create-top",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Let's give your group a name!",
                  },
                  "ref": null,
                  "rendered": "Let's give your group a name!",
                  "type": "h5",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "group-name-input",
                    "name": "groupName",
                    "onChange": [Function],
                    "placeholder": "group name",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "With SNAP NINJA, your group members get a bucket of snaps each week to give away to deserving teammates.",
              },
              "ref": null,
              "rendered": "With SNAP NINJA, your group members get a bucket of snaps each week to give away to deserving teammates.",
              "type": "h5",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h5>
                    How many snaps will your group members be able to give away?
                    <span
                                        className="join-span"
                    >
                                        (Between 1 - 500)
                    </span>
</h5>,
                  <input
                    className="points-input"
                    name="weeklyPoints"
                    onChange={[Function]}
                    placeholder="QTY"
                    value=""
/>,
                ],
                "className": "create-bottom",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "How many snaps will your group members be able to give away?",
                      <span
                        className="join-span"
>
                        (Between 1 - 500)
</span>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "How many snaps will your group members be able to give away?",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "(Between 1 - 500)",
                        "className": "join-span",
                      },
                      "ref": null,
                      "rendered": "(Between 1 - 500)",
                      "type": "span",
                    },
                  ],
                  "type": "h5",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "points-input",
                    "name": "weeklyPoints",
                    "onChange": [Function],
                    "placeholder": "QTY",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "success-message",
              },
              "ref": null,
              "rendered": "",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "CREATE",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "CREATE",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
